File: android/app/src/main/java/com/xirigo/ecommerce/feature/home/presentation/state/HomeUiState.kt
package com.xirigo.ecommerce.feature.home.presentation.state

import androidx.compose.runtime.Stable
import com.xirigo.ecommerce.feature.home.domain.model.DailyDeal
import com.xirigo.ecommerce.feature.home.domain.model.FlashSale
import com.xirigo.ecommerce.feature.home.domain.model.HomeBanner
import com.xirigo.ecommerce.feature.home.domain.model.HomeCategory
import com.xirigo.ecommerce.feature.home.domain.model.HomeProduct

@Stable
sealed interface HomeUiState {
    data object Loading : HomeUiState
    data class Success(val data: HomeScreenData) : HomeUiState
    data class Error(val message: String) : HomeUiState
}

@Stable
data class HomeScreenData(
    val banners: List<HomeBanner>,
    val categories: List<HomeCategory>,
    val popularProducts: List<HomeProduct>,
    val dailyDeal: DailyDeal?,
    val newArrivals: List<HomeProduct>,
    val flashSale: FlashSale?,
    val wishedProductIds: Set<String>,
)

File: ios/XiriGoEcommerce/Feature/Home/Presentation/State/HomeUiState.swift
import Foundation

// MARK: - HomeUiState

/// Represents the UI state of the home screen.
enum HomeUiState: Equatable {
    case loading
    case success(data: HomeScreenData)
    case error(message: String)
}

// MARK: - HomeScreenData

/// All data needed to render the home screen in success state.
struct HomeScreenData: Equatable {
    let banners: [HomeBanner]
    let categories: [HomeCategory]
    let popularProducts: [HomeProduct]
    let dailyDeal: DailyDeal?
    let newArrivals: [HomeProduct]
    let flashSale: FlashSale?
    let wishedProductIds: Set<String>
}
